### PR TITLE
Add project

### PR DESCRIPTION
**Pull request by AndrewOsypenko**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,1669 @@
+{
```
Comment (akoidan) [inline package-lock.json]: use yarn

File: src/app.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const express = require('express')
```
Comment (akoidan) [inline src/app.js]: https://nodejs.org/en/about/releases/
https://nodejs.org/api/esm.html#esm_import_statements

File: src/router/task.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+const express = require('express')
```
Comment (akoidan) [inline src/router/task.js]: Why express is imported here? can you import only router?

File: src/router/task.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+const Task = require('../models/task')
+const router = new express.Router()
+
+router.post('/', async (req, res) => {
+    const todo = new Task(req.body)
+    todo.save().then(() => {
```
Comment (akoidan) [inline src/router/task.js:7]: async/await, avoid callback hell

File: src/router/task.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        }).catch((e) => {
+        res.status(400).send(e)
+    })
+})
+
+router.get('/:id', async (req, res) => {
+    try {
+        const task = await Task.findOne({ _id: req.params.id })
+
+        if (!task) {
```
Comment (akoidan) [inline src/router/task.js]: Google if you express/mongoos can do that for you. If not can you decouple this?

File: src/router/task.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            return res.status(404).send()
+        }
+        res.send(task)
+    } catch (e) {
```
Comment (akoidan) [inline src/router/task.js]: google if express can do it for you. Decouple if not.

File: src/router/task.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        res.status(500).send()
+    }
+})
+
+router.patch('/:id', async (req, res) => {
+    const updates = Object.keys(req.body)
+    const allowedUpdates = ['title', 'description', 'completed']
+    const isValidOper = updates.every((update) => allowedUpdates.includes(update))
+
+    if (!isValidOper) {
```
Comment (akoidan) [inline src/router/task.js]: do we have proper exception handling? it's duplicated all over. `{error: string} `is used here. Another person will throw `{error: {details: "invalid updates}}`

File: src/router/task.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        return  res.status(401).send({error: 'Invalid updates!'})
+    }
+    try {
+        const task = await Task.findOne({ _id: req.params.id})
+        if (!task) {
+            return res.status(404).send(task)
+        }
+        updates.forEach((update) => task[update] = req.body[update])
```
Comment (akoidan) [inline src/router/task.js]: Think if this structure can be replaced with something else.

File: src/router/task.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
+const express = require('express')
+const Task = require('../models/task')
+const router = new express.Router()
+
+router.post('/', async (req, res) => {
```
Comment (akoidan) [inline src/router/task.js]: What happens with url when you add another entity? Check [REST-API-Design-Rulebook](https://pepa.holla.cz/wp-content/uploads/2016/01/REST-API-Design-Rulebook.pdf)

File: src/router/task.js (hunk 3 above)
Comment (AndrewOsypenko) [reply src/router/task.js]: Importing express needed to get Router method

Comment (akoidan) [reply src/router/task.js]: So why do you import express when you need Router?
```const {Router} = require("express");```

Comment (akoidan) [reply src/router/task.js]: https://en.wikipedia.org/wiki/Interface_segregation_principle

File: src/router/task.js (hunk 7 above)
Comment (AndrewOsypenko) [reply src/router/task.js]: Task.findOneAndUpdate({_id: req.params.id}, req.body, {}, (err, doc) => {
        if (err) {
            res.status(404).send(err)
        }
        res.send(doc)
    })

File: src/router/task.js (hunk 8 above)
Comment (AndrewOsypenko) [reply src/router/task.js]: Task.findOneAndUpdate({_id: req.params.id}, req.body, {}, (err, doc) => {
if (err) {
res.status(404).send(err)
}
res.send(doc)
})

File: src/router/task.js (hunk 6 above)
Comment (AndrewOsypenko) [reply src/router/task.js]: Task.findById({_id: req.params.id}, (err, doc) => {
        if (err) {
            res.status(404).send(err)
        }
        res.send(doc)
    })

Comment (akoidan) [reply src/router/task.js]: still bad, too much of code that's not related to the method business logic.

File: src/router/task.js (hunk 7 above)
Comment (akoidan) [reply src/router/task.js]: this should be in a single place, rather than a duplication in every method

File: src/router/task.js (hunk 8 above)
Comment (akoidan) [reply src/router/task.js]: I commented this line ` updates.forEach((update) => task[update] = req.body[update])` Can you simplify it?

**Pull request closed without merging**